задание_4.py

Poison1488 opened the pull request:
оказалось легко и просто)))

Poison1488 committed:
задание_4.py

оказалось легко и просто)))

wandrounik commented on task1/задание_4.py:
```diff
@@ -0,0 +1,16 @@
+try:
+    list = list(map(int,input("Введите список элементов:").split()))
+    number = int(input("Введите число:"))
+    operations_cod = int(input("Введите код операции:"))
```
не хватает проверки....что будет, если код операции будет больше 1 или меньше 0?

wandrounik replied:
видимо хотел написать operation_code

Poison1488 committed:
задание_2.py

так лучше? Или все таки лучше было другим способом сделать,я мог сделать с помощью циклов и списков,но подумал о том,что когда ты мне давно давал это задание,я о списках еще не знал

Poison1488 committed:
задание_4.py

исправил недочеты

wandrounik commented on task1/задание_2.py:
```diff
@@ -1,21 +1,27 @@
-            counter = user_variable
-            while counter >= 0:
-                print(counter,end = '.')
+        if max_range % 2 != 0:
```
Получается, что у  пользователя не получится вывести диапазон нечетных чисел от 0 до 10 например, потому что 10 четное? Очень странно получается (похожее и с нечетными).

wandrounik commented on task1/задание_4.py:
```diff
@@ -1,16 +1,18 @@
                 print (tmp,end = ' ')
     else:
-        if operations_cod == 0:
+        if operation_cod == 0:
```
есть конструкция elif

wandrounik commented on task1/задание_4.py:
```diff
@@ -1,16 +1,18 @@
     list = list(map(int,input("Введите список элементов:").split()))
     number = int(input("Введите число:"))
-    operations_cod = int(input("Введите код операции:"))
+    operation_cod = int(input("Введите код операции:"))
```
operation_cod**e**
английская грамматика!!! (не cod, а code)

wandrounik commented on task1/задание_4.py:
```diff
@@ -1,16 +1,18 @@
\ No newline at end of file
+                    print (tmp,end = ' ')
+        else:
+            print("Вы ввели не верный тип данных")
```
Я так понимаю, что ошибка в неправильном коде операции, а не в типе данных

Poison1488 committed:
задание_2.py

так лучше смотрится)

wandrounik commented on task1/задание_2.py:
```diff
@@ -1,27 +1,23 @@
-                counter -= 2
-        else:
-            print("Вы ввели не верный максимальный диапазон для данного кода операции")
+    list = []
```
Тебе по сути и не нужно сохранять это в список....можно в цикле for проверять и сразу выводить

wandrounik commented on task1/задание_2.py:
```diff
@@ -1,27 +1,23 @@
         else:
-            print("Вы ввели не верный код операции")
\ No newline at end of file
+            if operation_code == 1:
```
Опять же можно использовать elif

Poison1488 committed:
задание_2.py

исправил все косяки

Poison1488 committed:
задание_3.py

исправил все косяки

Poison1488 committed:
задание_4.py

исправил все косяки

wandrounik commented on task1/задание_2.py:
```diff
@@ -5,19 +5,13 @@
-    for tmp in range(min_range,max_range):
-        if tmp != max_range:
-            list.append(tmp)
+    if operation_code == 0:
```
Код можно еще сократить в два раза...например: тебе не нужно использовать два цикла for, можно в одном цикле проверять твои условия (что-то вроде этого): 
_if operation_code == 0 **and** tmp % 2 == 0:_

Poison1488 committed:
Merge branch 'korenandr:main' into main

Poison1488 committed:
задание_2.py

все исправил

Poison1488 committed:
задание_5.py

все исправил

wandrounik commented on task1/задание_5.py:
```diff
@@ -0,0 +1,17 @@
+else:
+    number_of_elements = len(list)
+    final_value = number_of_elements - 1
+    difference = list[1] - list[0]
```
Что будет, если введу одно число?

wandrounik commented on task1/задание_5.py:
```diff
@@ -0,0 +1,17 @@
+    difference = list[1] - list[0]
+    for number in range(1,number_of_elements):
+        previous_index = number - 1
+        if list[number] - list[previous_index] == difference:
```
Можно сделать проще:
1. Проверить != difference (в этом случае не является арифм прогрессией и сделать return)
2. Если весь список прошелся и не выполнился первый пункт, следовательно все хорошо и можно после цикла просто вывести сообщение (что является арифм прогрессией)

Poison1488 committed:
Merge branch 'korenandr:main' into main
